WalletWasabi.Gui/Controls/WalletExplorer/SendControlViewModel.cs

```diff
@@ -786,13 +824,45 @@ public ErrorDescriptors ValidateAddress()
 			return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "Invalid address."));
 		}
 
-		[ValidateMethod(nameof(ValidateAddress))]
+		public ErrorDescriptors ValidateCustomChangeAddress()
+		{
+			if (string.IsNullOrWhiteSpace(CustomChangeAddress))
+			{
+				return ErrorDescriptors.Empty;
+			}
+
+			if (IsMax)
+			{
+				return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "Spending whole coins does not generate change."));
+			}
+
+			if (Address?.Trim() == CustomChangeAddress.Trim())
+			{
+				return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "The active address and the change addresses cannot be the same."));
+			}
+
+			if (AddressStringParser.TryParseBitcoinAddress(CustomChangeAddress, Global.Network, out _))
+			{
+				return ErrorDescriptors.Empty;
+			}
+
+			return new ErrorDescriptors(new ErrorDescriptor(ErrorSeverity.Error, "Invalid change address."));
+		}
+
+		[ValidateMethod(nameof(ValidateActiveAddress))]
 		public string Address
 		{
 			get => _address;
 			set => this.RaiseAndSetIfChanged(ref _address, value);
 		}
 
+		[ValidateMethod(nameof(ValidateCustomChangeAddress))]
+		public string CustomChangeAddress
+		{
+			get => _customChangeAddress;
+			set => this.RaiseAndSetIfChanged(ref _customChangeAddress, value);
```

Comment:
In some places we use `CustomChangeAddress.Trim()` while in other we use simple `CustomChangeAddress`. I think we could eliminate the redundancy while being more consistent if we do something like this:

```suggestion
			set => this.RaiseAndSetIfChanged(ref _customChangeAddress, value?.Trim());
```

Reply:
I like this. We could do the same with Address.

Reply:
Note this prevents spaces to be written to the fields. For example if I write in a Bitcoin URL and I want space in the label, then I will only be able to achieve this if I copypaste the Bitcoin URL.  

Also this needs to be tested on OSX, it may result in freeze. Anyway, I implemented it, good idea.